src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -1725,36 +1763,36 @@ public override IOperation VisitInvocation(IInvocationOperation operation, int?
             }
 
             ImmutableArray<IArgumentOperation> arguments = operation.Arguments;
-            foreach (IArgumentOperation argument in arguments)
-            {
-                _evalStack.Push(Visit(argument.Value));
-            }
-
-            ImmutableArray<IArgumentOperation> visitedArguments;
-            if (!arguments.IsEmpty)
-            {
-                var builder = ArrayBuilder<IArgumentOperation>.GetInstance();
-                for (int i = arguments.Length - 1; i >= 0; i--)
-                {
-                    IOperation visitedValue = _evalStack.Pop();
-                    var originalArgument = (BaseArgument)arguments[i];
-                    builder.Add(new ArgumentOperation(visitedValue, originalArgument.ArgumentKind, originalArgument.Parameter,
-                                                      originalArgument.InConversionConvertibleOpt, originalArgument.OutConversionConvertibleOpt,
-                                                      semanticModel: null, originalArgument.Syntax, originalArgument.ConstantValue, originalArgument.IsImplicit));
-                }
-                builder.ReverseContents();
-                visitedArguments = builder.ToImmutableAndFree();
-            }
-            else
-            {
-                visitedArguments = ImmutableArray<IArgumentOperation>.Empty;
-            }
-
+            PushArray(arguments, arg => arg.Value);
+            ImmutableArray<IArgumentOperation> visitedArguments = PopArray(arguments, RewriteArgumentFromArray);
             IOperation visitedInstance = operation.Instance == null ? null : _evalStack.Pop();
 
             return new InvocationExpression(operation.TargetMethod, visitedInstance, operation.IsVirtual, visitedArguments, semanticModel: null, operation.Syntax, operation.Type, operation.ConstantValue, operation.IsImplicit);
         }
 
+        public override IOperation VisitObjectCreation(IObjectCreationOperation operation, int? captureIdForResult)
+        {
+            ImmutableArray<IArgumentOperation> arguments = operation.Arguments;
+            PushArray(arguments, arg => arg.Value);
+            ImmutableArray<IArgumentOperation> visitedArgs = PopArray(arguments, RewriteArgumentFromArray);
```

Comment:
>RewriteArgumentFromArray [](start = 81, length = 24)

Consider caching this delegate in a field and reusing in all sites.

Reply:
Fixed.

---
In reply to: [171975952](https://github.com/dotnet/roslyn/pull/25178#discussion_r171975952) [](ancestors = 171975952)